src/transformers/training_args.py

```diff
@@ -1687,6 +1687,16 @@ def __post_init__(self):
                 mixed_precision_dtype = "bf16"
             os.environ["ACCELERATE_MIXED_PRECISION"] = mixed_precision_dtype
 
+        # Finally set the `TrainingArguments` to be immutable
+        self._frozen = True
+
+    def __setattr__(self, name, value):
+        # Once fully through the `__post_init__`, `TrainingArguments` are immutable
+        if getattr(self, "_frozen", False):
```

Comment:
Since adding `_frozen` to the dataclass args earlier would make it show up as an available option to pass through, we don't set it until the very end of `__post_init__`, hence the `getattr` instead of setting it earlier.